/*
 * Copyright (c) 2017 Broadcom. The term "Broadcom" refers
 * to Broadcom Limited and/or its subsidiaries.

 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.

 * You may obtain a copy of the License at
 * http://www.apache.org/licenses/LICENSE-2.0

 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

syntax = "proto2";

import "sai_tam_event_flow_learning.proto";
import "sai_tam_event_threshold_breach.proto";
import "sai_tam_buffer_stats.proto";

message Event {
    required uint64 timestamp                 = 1;

    oneof EventType {
        Drop         drop_event               = 2;
        FlowLearning flow_event               = 3;
        ThresholdBreach threshold_event       = 4;
        SwitchBufferStats switch_buffer_event = 5;
    }
}

message Drop {
    optional uint32 dummy           = 1;
}

message GenEvent {
    required string system_id       = 1;
    optional uint32 component_id    = 2;
    optional uint32 sub_component_id= 3;
    repeated EventPair eventpair    = 4;
    optional string hostname        = 5;
}

message EventPair {
    repeated Event event            = 1;
    repeated Flow flow              = 2;
}
